staging/src/k8s.io/code-generator/cmd/client-gen/generators/util/tags.go

```diff
@@ -146,16 +242,96 @@ func ParseClientGenTags(lines []string) (Tags, error) {
 		}
 		ret.SkipVerbs = skipVerbs
 	}
+	var err error
+	if ret.Extensions, err = parseClientExtensions(values); err != nil {
+		return ret, err
+	}
 	return ret, validateClientGenTags(values)
 }
 
+func parseClientExtensions(tags map[string][]string) ([]extension, error) {
+	var ret []extension
+	for name, values := range tags {
+		if !strings.HasPrefix(name, genClientPrefix+"method") {
+			continue
+		}
+		for _, value := range values {
+			// the value comes in this form: "Foo,verb=create"
+			ext := extension{}
+			parts := strings.Split(value, ",")
+			if len(parts) == 0 {
+				return nil, fmt.Errorf("invalid of empty extension verb name: %q", value)
+			}
+			// The first part represents the name of the extension
```

Comment:
I don't like forcing order of sub elements. Issue to make this more flexible please.

Reply:
the //+genclient:method=NAME is the required token we looking for, the ordering of the following arguments should not be forced (as you noted in comment bellow).